numpy/core/numeric.py

```diff
@@ -2111,8 +2167,16 @@ def identity(n, dtype=None):
            [0.,  0.,  1.]])
 
     """
+    if like is not None:
+        return _identity_with_like(n, dtype=dtype, like=like)
+
     from numpy import eye
-    return eye(n, dtype=dtype)
+    return eye(n, dtype=dtype, like=like)
```

Comment:
This one is a really strong argument for not having a like override -  it shouldn't be up to dask or cupy to decide whether `eye` and `identity` are aliases.

Reply:
(also, this `like=like` is always `like=None`)

Reply:
I'll fix that in a separate PR, thanks @eric-wieser for spotting this.